src/node.ts

```diff
@@ -29,7 +37,7 @@ export abstract class BaseNode<
   Kind extends FunctionlessNode["kind"],
   Parent extends FunctionlessNode | undefined = FunctionlessNode | undefined
 > {
-  abstract readonly nodeKind: "Err" | "Expr" | "Stmt" | "Decl";
+  abstract readonly nodeKind: "Err" | "Expr" | "Stmt" | "Decl" | "Node";
```

Comment:
What is "Node"? That doesn't seem right, we only have errors, expressions, statements and declarations.

Reply:
What is `ObjectBindingPattern` then? In `ts`, it is just a `Node`.

```ts
    export interface ObjectBindingPattern extends Node {
        readonly kind: SyntaxKind.ObjectBindingPattern;
        readonly parent: VariableDeclaration | ParameterDeclaration | BindingElement;
        readonly elements: NodeArray<BindingElement>;
    }
```

`Node` is a utility entity which is part of the tree, but only existings in specific contects.

Object and Array Binding:
* Do not declare (the `BindingElm` does)
* Are not a statement, cannot existing in a block
* Are not an expression, cannot exist in a ExprStmt, etc.
* Not an error, obviously

Reply:
When we add `Switch` (#306 ), case and default are other cases.
```ts
    export interface CaseBlock extends Node {
        readonly kind: SyntaxKind.CaseBlock;
        readonly parent: SwitchStatement;
        readonly clauses: NodeArray<CaseOrDefaultClause>;
    }
```

Reply:
Catch also fits this description.

```ts
    export interface CatchClause extends Node {
        readonly kind: SyntaxKind.CatchClause;
        readonly parent: TryStatement;
        readonly variableDeclaration?: VariableDeclaration;
        readonly block: Block;
    }
```